### PR TITLE
libopae: Generate config_int.h with version info

### PR DESCRIPTION
**Pull request by luebbers**
Creates a config_int.h containing CMake variables, including version
information and the current git hash.

This is in preparation for an API that allows software to query the
current OPAE API's release.

### CODE REVIEW

File: cmake/config/config.h.in [hunk 1]
```diff
@@ -218,3 +218,6 @@
 /* Define to the type of an unsigned integer type wide enough to hold a
    pointer, if such a type exists, and if the system does not define it. */
 #cmakedefine uintptr_t
+
+#cmakedefine INTEL_FPGA_API_VERSION @INTEL_FPGA_API_VERSION@
```
Comment (r-rojo) [inline cmake/config/config.h.in]: This doesn't need quotes too?

File: CMakeLists.txt [hunk 2]
```diff
@@ -27,6 +27,23 @@
...
+find_program(GIT_EXECUTABLE git)
+if(EXISTS ${GIT_EXECUTABLE})
+  execute_process(COMMAND ${GIT_EXECUTABLE} log -1 --format=%h
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    OUTPUT_VARIABLE GIT_COMMIT_HASH
+    RESULT_VARIABLE GIT_LOG_RESULT
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(NOT ${GIT_LOG_RESULT} EQUAL 0)
+      set(GIT_COMMIT_HASH unknown)
+    endif(NOT ${GIT_LOG_RESULT} EQUAL 0)
```
Comment (r-rojo) [inline CMakeLists.txt]: What do we see when we do a `git archive`? Should we have this fall back to using something like the `id` you had in another PR?

File: CMakeLists.txt [hunk 3]
```diff
@@ -27,6 +27,23 @@
 cmake_minimum_required(VERSION 2.8.11)
 project(opae)
 
+############################################################################
+## Get git hash info, if available #########################################
+############################################################################
```
Comment (r-rojo) [inline CMakeLists.txt]: Should we make this a macro?

Comment (luebbers) [reply CMakeLists.txt]: It wasn't one before and is only used once.

File: CMakeLists.txt (hunk 2 above)
Comment (luebbers) [reply CMakeLists.txt]: Yes, we should. I'm going to add that in another PR.

File: cmake/config/config.h.in (hunk 1 above)
Comment (luebbers) [reply cmake/config/config.h.in]: You're right, it does. I'll also add numeric constants for the individual components (major, minor, revision).

Comment (r-rojo) [reply cmake/config/config.h.in]: What about a struct? Can we initialize a static struct with the version numbers?

**Pull request merged**